Triangle Streaming Server/ReceiveStream.cs

```diff
@@ -20,33 +56,41 @@ protected override void OnOpen()
 
 		protected override void OnMessage(MessageEventArgs e)
 		{
-			if(!e.Data.Equals("PUBKEY"))
+			if(e.Data.Equals("PUBKEY"))
 			{
-				return;
+				//Send public key to client
+				Send(_pubKey);
 			}
-
-			if (_keyPair == null)
+			else if(e.Data.StartsWith("WATCH "))
 			{
-				//Read privatekey.pem from executable directory
-				using (var reader = File.OpenText(AppDomain.CurrentDomain.BaseDirectory + @"\privatekey.pem"))
+				//Get the ID specified in the WATCH command (for example, "WATCH 158sadsa489g")
+				string streamToWatch = e.Data.Remove(0, 6);
+
+				if (Clients.ContainsKey(this.ID))
 				{
-					_keyPair = (AsymmetricCipherKeyPair)new PemReader(reader).ReadObject();
+					Clients.Remove(this.ID);
 				}
-			}
 
-			//Get public key string
-			if(_pubKey == null)
+				Clients.Add(this.ID, streamToWatch);
+			}
+			else if(e.Data.Equals("LIST"))
 			{
-				TextWriter textWriter = new StringWriter();
-				PemWriter pemWriter = new PemWriter(textWriter);
-				pemWriter.WriteObject(_keyPair.Public);
-				pemWriter.Writer.Flush();
-				_pubKey = textWriter.ToString();
-				textWriter.Close();
+				//Send a list of streams to the client.
+				JArray streams = new JArray();
```

Comment:
Cant we use a JSON serializer. Like [Newtonsoft JSON](http://www.newtonsoft.com/json )